models/library.py

```diff
@@ -49,7 +58,8 @@ class RentalLog(db.Model):
                         nullable=False)
     _borrow_time = db.Column(db.DateTime)
     _return_time = db.Column(db.DateTime)
-    reserved = db.Column(db.Boolean)
+    book_status = db.Column(ChoiceType(BookStatus, impl=db.Integer()))
+    _reservation_timestamp = db.Column(db.DateTime)
     returned = db.Column(db.Boolean)
```

Comment:
I think we can get rid off `returned` field here, because now we have `BookStatus` with that field :)